opencog/nlp/microplanning/main.scm

```diff
@@ -92,6 +92,13 @@
 	; Initialize the sentence forms as needed
 	(microplanning-init)
 
+    ; Reset (clear) SuReal cache to assure it is empty before start calling sureal.
+    ; Remark that SuReal Cache is not thread-safe. So it is not supposed to be used 
```

Comment:
Because in this version there is only one instance of the cache (reached via a singleton wrapper). So if two threads with two different Microplanner queries add stuff to the cache, one may lead to false hits in the other. In addition to this, the cache is reset (clear) just before the Microplanner query starts so if the second query reach the reset point before the first query ended, the cache will be reset during the lifetime of the first query, which may lead to false misses.

If this cache approach proves to be interesting (valuable speedup), I will implement a second version where each Microplanner query have its own cache instance.